server/bin/test_data_importer.py

```diff
@@ -0,0 +1,865 @@
+        keys = ['branding', 'contractNumber', 'accountNumber', 'orderNumber', 'subscriptionId', 'upstreamPoolId']
+        safe_merge(pool_data, pool, include=keys)
+
+        if self.hosted_test:
+            # Need to set a subscription ID if we're generating this upstream
+            pool['id'] = random_id('upstream')
+
+            # Product is an object in hosted test mode
+            pool['product'] = { 'id': pool_data['product_id'] }
+
+            # The owner must be set in the subscription data for upstream subscriptions
+            pool['owner'] = { 'key': owner_key }
+
+            endpoint = 'hostedtest/subscriptions'
+        else:
+            # Copy over the pool (special case because hosted test data is different)
+            pool['productId'] = pool_data['product_id']
+
+            # Generate fields that make this look like an upstream subscription
+            if not 'subscriptionId' in pool:
+                pool['subscriptionId'] = random_id('srcsub')
+
+            if 'subscription_subkey' in pool_data:
+                pool['subscriptionSubKey'] = pool_data['subscription_subkey']
+            elif 'subscriptionSubKey' in pool_data:
+                pool['subscriptionSubKey'] = pool_data['subscriptionSubKey']
+            else:
+                pool['subscriptionSubKey'] = 'master'
+
+            if not 'upstreamPoolId' in pool:
+                pool['upstreamPoolId'] = random_id('upstream')
+
+            endpoint = 'owners/{owner_key}/pools'.format(owner_key=owner_key)
+
+        return self.post(endpoint, {}, pool).json()
+
+    def list_pools(self, owner_key):
+        endpoint = 'owners/{owner_key}/pools'.format(owner_key=owner_key)
+        return self.get(endpoint).json()
+
+    def create_activation_key(self, owner_key, key_data):
+        key = safe_merge(key_data, {})
+        return self.post('owners/{owner_key}/activation_keys'.format(owner_key=owner_key), {}, key).json()
+
+    def add_pool_to_activation_key(self, key_id, pool_id, quantity=None):
+        endpoint = 'activation_keys/{key_id}/pools/{pool_id}'.format(key_id=key_id, pool_id=pool_id)
+        params = {}
+
+        if quantity is not None:
+            params['quantity'] = quantity
+
+        return self.post(endpoint, params).json()
+
+    def refresh_pools(self, owner_key, wait_for_completion=True):
+        endpoint = 'owners/{owner_key}/subscriptions'.format(owner_key=owner_key)
+        job_status = self.put(endpoint, { 'lazy_regen': True }, None).json()
+
+        if job_status and wait_for_completion:
+            finished_states = ['ABORTED', 'FINISHED', 'CANCELED', 'FAILED']
+
+            while job_status['state'] not in finished_states:
+                time.sleep(1)
+                job_status = self.get(job_status['statusPath']).json()
+
+            # Clean up job status
+            # self.delete('jobs', {'id': job_status['id']})
+
+        return job_status
+
+def create_owners(cp, data):
```

Comment:
create_owners, create_users, etc. only exist because of the previous request to break up the process_file function. They serve no purpose other than reducing the length of that function to make the linter happy.

The goal of the Candlepin class is to provide the API. The methods mentioned above here provide business logic that itself uses the API.

Reply:
OK, np.